cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -97,6 +130,26 @@ public void classify(CloseableIterator<LogEvent> logIter, HttpResponder responde
     List<ErrorClassificationResponse> responses = new ArrayList<>(responseMap.values());
     responses.addAll(responseSet);
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(responses));
+
+    // emit metric
+    MetricsContext metricsContext = metricsCollectionService.getContext(ImmutableMap.of(
+        Metrics.Tag.NAMESPACE, namespace,
+        Metrics.Tag.PROGRAM, programId,
+        Metrics.Tag.APP, appId,
+        Metrics.Tag.RUN_ID, runId));
+
+    for (ErrorClassificationResponse response : responses) {
+      MetricsContext context = metricsContext.childContext(ImmutableMap.of(
+          Metrics.Tag.ERROR_CATEGORY, response.getErrorCategory(),
+          Metrics.Tag.ERROR_TYPE, response.getErrorType(),
+          Metrics.Tag.DEPENDENCY, response.getDependency(),
+          Metrics.Tag.ERROR_CODE_TYPE, response.getErrorCodeType(),
+          Metrics.Tag.ERROR_CODE, response.getErrorCode()
+      ));
+      context.increment(Metrics.Program.FAILED_RUNS_CLASSIFICATION_COUNT, 1);
```

Comment:
Should you use gauge? cache is in memory and pod restart / cache overflow would mean doublecounting. And gauge should work the way you want (with all the context) unless I miss something.

Reply:
Updated the metric to gauge but I have still kept the cache so that we don't have to do re-processing of logs until cache is overflowed/ pod restart.